feature/gribi/ate_tests/ipv4_entry_test/ipv4_entry_test.go

```diff
@@ -0,0 +1,351 @@
+		t.Run(tc.desc, func(t *testing.T) {
+			gribic := dut.RawAPIs().GRIBI().Default(t)
+			c := fluent.NewClient()
+			c.Connection().WithStub(gribic).
+				WithRedundancyMode(fluent.ElectedPrimaryClient).
+				WithFIBACK().
+				WithInitialElectionID(1, 0)
+			c.Start(ctx, t)
+			defer c.Stop(t)
+			c.StartSending(ctx, t)
+			if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
+				t.Fatalf("Await got error during session negotiation: %v", err)
+			}
+
+			if tc.downPort != nil {
+				ate.Actions().NewSetPortState().WithPort(tc.downPort).WithEnabled(false).Send(t)
+				defer ate.Actions().NewSetPortState().WithPort(tc.downPort).WithEnabled(true).Send(t)
+			}
+
+			c.Modify().AddEntry(t, tc.entries...)
+			if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
+				t.Fatalf("Await got error for entries: %v", err)
+			}
+			defer func() {
+				c.Modify().DeleteEntry(t, tc.entries...)
+				if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
+					t.Fatalf("Await got error for entries: %v", err)
+				}
+			}()
+
+			for _, wantResult := range tc.wantOperationResults {
+				chk.HasResult(t, c.Results(t), wantResult, chk.IgnoreOperationID())
+			}
+
+			validateTrafficFlows(t, ate, tc.wantGoodFlows, tc.wantBadFlows)
+		})
+	}
+}
+
+// configureDUT configures port1-3 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := dut.Config()
+
+	p1 := dut.Port(t, "port1")
+	p2 := dut.Port(t, "port2")
+	p3 := dut.Port(t, "port3")
+
+	d.Interface(p1.Name()).Replace(t, dutPort1.NewInterface(p1.Name()))
+	d.Interface(p2.Name()).Replace(t, dutPort2.NewInterface(p2.Name()))
+	d.Interface(p3.Name()).Replace(t, dutPort3.NewInterface(p3.Name()))
+}
+
+// configreATE configures port1-3 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	p2 := ate.Port(t, "port2")
+	p3 := ate.Port(t, "port3")
+
+	atePort1.AddToATE(top, p1, &dutPort1)
+	atePort2.AddToATE(top, p2, &dutPort2)
+	atePort3.AddToATE(top, p3, &dutPort3)
+
+	top.Push(t).StartProtocols(t)
+
+	return top
+}
+
+// createFlow returns a flow from atePort1 to the dstPfx, expected to arrive on ATE interface dsts.
+func createFlow(name string, ate *ondatra.ATEDevice, ateTop *ondatra.ATETopology, dsts ...*attrs.Attributes) *ondatra.Flow {
+	hdr := ondatra.NewIPv4Header()
+	hdr.WithSrcAddress(dutPort1.IPv4).
+		DstAddressRange().WithMin(dstPfxMin).WithMax(dstPfxMax).WithCount(dstPfxCount)
+
+	endpoints := []ondatra.Endpoint{}
+	for _, dst := range dsts {
+		endpoints = append(endpoints, ateTop.Interfaces()[dst.Name])
+	}
+
+	flow := ate.Traffic().NewFlow(name).
+		WithSrcEndpoints(ateTop.Interfaces()[atePort1.Name]).
+		WithDstEndpoints(endpoints...).
+		WithHeaders(ondatra.NewEthernetHeader(), hdr)
+
+	return flow
+}
+
+func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good []*ondatra.Flow, bad []*ondatra.Flow) {
+	if len(good) == 0 && len(bad) == 0 {
```

Comment:
Do you need to add **t.Fatalf()** instead of return as traffic flow is required for all the test case?

Reply:
I was using no traffic for testing new cases -- I'm not sure that there is a requirement that every test case must have traffic.  This just prevents wasting time trying to send traffic when there is no traffic requested.

Reply:
SGTM.